### PR TITLE
Improve cipher upload UX

### PR DESCRIPTION
**Pull request by ChihChengLiang**
We show a progress bar for the uploading of 200MB server key share.
```
[00:00:00] ---------------------------------------- 0% 0 B/s Uploading... 
```

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -87,15 +90,39 @@ impl WebClient {
...
+                    ProgressStyle::with_template(
+                        "[{elapsed_precise}] {bar:40.cyan/blue} {percent}% {bytes_per_sec} {msg}",
+                    )
+                    .unwrap()
+                    .progress_chars("##-"),
+                );
+                bar.set_message("Uploading...");
+
+                // Create the ProgressReader
+                let reader = ProgressReader {
```
Comment (ed255) [inline src/client.rs]: Maybe do a constructor for ProgressReader?
Then we only pass body and chunk_size, and the rest is initialized internally.

File: src/client.rs [hunk 2]
```diff
@@ -194,3 +221,34 @@ async fn handle_response_test<T: Send + for<'de> Deserialize<'de> + 'static>(
...
+
+        let remaining = self.inner.len() - self.position;
+        let to_read = self.chunk_size.min(remaining.min(buf.remaining()));
+        let end = self.position + to_read;
+        buf.put_slice(&self.inner[self.position..end]);
+        self.position = end;
+
+        let end = buf.filled().len();
+        let new_bytes = (end - start) as u64;
+        self.bytes_read += new_bytes;
```
Comment (ed255) [inline src/client.rs]: `new_bytes` should be the same as `to_read` so the first two lines can be removed.
`self.bytes_read` is the same as position so we should also be able to remove it.

**Review by ed255 — APPROVED** [review]
I added to small improvements, otherwise LGTM!

File: src/client.rs (hunk 1 above)
Comment (ChihChengLiang) [reply src/client.rs]: Good idea. Resolved in [296bed4](https://github.com/ChihChengLiang/karma_calculator/pull/23/commits/296bed4f9230ab5aca3d3cc5065463e38d458365)

File: src/client.rs (hunk 2 above)
Comment (ChihChengLiang) [reply src/client.rs]: Resolved in [296bed4](https://github.com/ChihChengLiang/karma_calculator/pull/23/commits/296bed4f9230ab5aca3d3cc5065463e38d458365) too

**Pull request merged**